presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -518,6 +520,22 @@ public Boolean visitParameter(Parameter node, Void context)
             return process(parameters.get(node.getPosition()), context);
         }
 
+        public Boolean visitGroupingOperation(GroupingOperation node, Void context)
+        {
+            List<Expression> argumentsNotInGroupBy = node.getGroupingColumns().stream()
+                    .filter((argument) -> !expressions.contains(argument))
+                    .collect(toImmutableList());
+            if (!argumentsNotInGroupBy.isEmpty()) {
+                throw new SemanticException(
+                        INVALID_PROCEDURE_ARGUMENTS,
+                        node,
+                        "Arguments in GROUPING() must be in the query's GROUP BY columns. Remove columns '%s' from GROUPING() or add them to the GROUP BY.",
```

Comment:
I dislike having only `The arguments to GROUPING must be expressions referenced by the query's GROUP BY` because it implies that `grouping()` can reference `GROUP BY` columns in a sub-query, which it can't. Even if "associated query level" is not used by Presto I think any SQL user should be able to deduce what that means.